README.md

```diff
@@ -65,6 +79,87 @@ To generate coverage results, set the `mapCoverage` property in the `jest` confi
 
 > Please note that the `outDir` property in the `jest` configuration section is removed in coverage mode, due to [#201](https://github.com/kulshekhar/ts-jest/issues/201).
 
+## Sensible Defaults
+`ts-jest` tries to ship with sensible defaults, to get you on your feet as quickly as possible.
+
+### Sourcemap support
+Sourcemaps should work out of the box. That means your stack traces should have the correct line numbers,
+and you should be able to step through the TypeScript code using a debugger.
+
+### Automatically finds tsconfig.json
+`ts-jest` automatically located your `tsconfig` file.
+If you want to compile typescript with a special configuration, you [also can](#tsconfig)
+
+### Supports synthetic modules 
+If you're on a codebase where you're using synthetic default imports, e.g. 
+```javascript 1.6
+//Regular imports
+import * as React from 'react';
+
+//Synthetic default imports:
+import React from 'react';
+```
+`ts-jest` tries to support that. If `allowSyntheticDefaultImports` is set to true in your `tsconfig` file, it uses babel
+to automatically create the synthetic default exports for you - nothing else needed. 
+You can opt-out of this behaviour with the [skipBabel flag](#skipping-babel)
+
+### Supports automatic of jest.mock() calls
+[Just like Jest](https://facebook.github.io/jest/docs/manual-mocks.html#using-with-es-module-imports) `ts-jest`
+automatically hoists your `jest.mock()` calls to the top of your file.
+You can opt-out of this behaviour with the [skipBabel flag](#skipping-babel)
+
+## Configuration
+Sometimes you just need a little extra configurability. 
+
+### tsconfig
+By default this package will try to locate `tsconfig.json` and use its compiler options for your `.ts` and `.tsx` files.
+
+You are able to override this behaviour and provide another path to your config for TypeScript by using `__TS_CONFIG__` option in `globals` for `jest`:
+```json
+{
+  "jest": {
+    "globals": {
+      "__TS_CONFIG__": "my-tsconfig.json"
+    }
+  }
+}
+```
+Or even declare options for `tsc` instead of using separate config, like this:
+```json
+{
+  "jest": {
+    "globals": {
+      "__TS_CONFIG__": {
+        "module": "commonjs",
+        "jsx": "react"
+      }
+    }
+  }
+}
+```
+For all available `tsc` options see [TypeScript docs](https://www.typescriptlang.org/docs/handbook/compiler-options.html).
+
+Note that if you haven't explicitly set the `module` property in the `__TS_CONFIG__` setting (either directly or through a separate configuration file), it will be overwritten to `commonjs` (regardless of the value in `tsconfig.json`) since that is the format Jest expects. This only happens during testing.
+
+### Skipping Babel
+If you want to skip the babel transpilation step in ts-jest, you can. This means `jest.mock()` calls will not be hoisted to the top,
```

Comment:
We should add that ts-jest uses babel by default so that `mock` statements are hoisted properly. That would make it clear why users might want to skip babel (if they don't use mocks)